backend/apis/parse-docket-pdf.utils.js

```diff
@@ -89,13 +89,80 @@ function parseCharges(lines, sections) {
 }
 
 function parseAccountSummary(lines, sections) {
-  const accounts = [];
+  let subsections = [];
+  let balances = [];
+  const sectionLastIndex = sections.filter(section => {
+    return section.name === "PROCEEDINGS" && section.lineNumber;
+  })[0].lineNumber;
+
+  function addAccounts(line, index) {
+    const words = line.split(/\s/);
+    const isSubsection =
+      words.length > 0 &&
+      words[0].toUpperCase() === words[0] &&
+      !words.includes("CASE") && // prevents CASE NUMBER lines from defining a subsection
+      words[1]; // prevents line new page diving line from defining a subsection;
+
+    if (isSubsection) {
+      subsections.push({ name: line, lineNumber: index });
+    }
+
+    balances = subsections
+      .map((subsection, arrayIndex) => {
+        const endIndex =
+          arrayIndex + 1 !== subsections.length
+            ? subsections[arrayIndex + 1].lineNumber
+            : sectionLastIndex;
+
+        let amountDue,
+          amountPaid,
+          costType,
+          collection,
+          originalAmountDue,
+          amendedAmountDue;
+
+        for (let i = subsection.lineNumber; i < endIndex; i++) {
+          const line = lines[i];
+          const words = line.split(/\s/);
+
+          if (line.includes("Amount Due")) {
+            if (line.includes("Original")) {
+              originalAmountDue = words[words.length - 1];
+            } else if (line.includes("Amended")) {
+              amendedAmountDue = words[words.length - 1];
+            }
+            amountDue = words[words.length - 1];
+          } else if (line.includes("Amount Paid") || line.includes("Paid In")) {
+            amountPaid = words[words.length - 1];
+          }
+          // console.log(line);
+          costType = getCostType(line) || costType;
+          collection = line.includes("State Debt Collection");
+        }
+        const returnObj = {
+          name: subsection.name,
+          amountDue,
+          originalAmountDue,
+          amendedAmountDue,
+          amountPaid,
+          costType,
+          collection
+        };
+        return amountDue && returnObj;
+      })
+      .filter(result => {
+        return result;
+      });
+  }
+  function getCostType(line) {
```

Comment:
Can we make this function better to include detection of fines and fees (in addition to restitution)? For example, this line of code is for a fine:

https://github.com/JustUtahCoders/utahexpungements.org/blob/c9522469b27675264611ce65654147627a1566bd/backend/apis/docket-pdf-test-files/account-summary.js#L53

Fines, fees, and restitution are treated differently when determining eligibility, which is why it's important to try to determine what kind of charge it is, if possible.

Reply:
I'll look into it.

Reply:
Should I add a value to costType for every case? That's a pretty straighforward task...

Reply:
We should return a cost type whenever we can derive one.